### PR TITLE
Implemented persistent run times

### PR DESCRIPTION
**Pull request by laubed**
It's usefull when running jobs with every() because jobs won't forget their last time they ran.
Jobs now create a persistent file (only if specified). It also needs an identifier for each job.

### CODE REVIEW

**Comment by dbader** [comment]
Linking this to https://github.com/dbader/schedule/issues/17 for reference.

**Comment by dbader** [comment]
Thanks for the PR 😃  Adding persistence to schedule has been a point of discussion for a while (for example see #17). Doing a robust job persistence implementation will be a lot of work. I recommend checking out the section on job storage backends in the [APScheduler docs](https://apscheduler.readthedocs.io/en/latest/).

Some feedback on this implementation:

* What if we're running in an environment where we can't write to disk? (really likely for web apps running on PaaS for example)
* Will this work on non-UNIX OSs?
* What if the job store gets corrupted (can't parse the timestamp)?
* What about concurrent updates to the job store? Do we need to handle that?

Not trying to be rude here or to discourage you but I honestly think adding persistence properly and in a robust way would be a massive effort and it would be tough to do it in a generic enough way...

**Comment by laubed** [comment]
I think we talk about different scopes of persistence. The PR is meant to provide the correct timings for a given/known ca. It does explicitly not save calls so they can be loaded via a simple load() function.
1. It is indeed a serious flaw to only use files as a storage solution. I will investigate that further.
2. It uses cross platform code out of standard python libraries. I see no reason why it shouldn't work. Hoing go test at least windows later today.
3. We can't do anything against corruption, so we need to agree to a policy on that (either don't run the job at all or run it as if there was no persistence data)
I admit that i didn't thought about that issue.
4. Haven't thought about that one as well. But i would assume that there should not be concurrent access because: (1) one instance processes all tasks sequentially, (2) two or more instances should not share a persistence storage (otherwise we need to develop a task orchestration tool instead of a scheduler which is out of scope of this project)



Daniel Laube | Harzblick 48 | 38122 Braunschweig | +49 176 46669865

Am 7. November 2016 14:01:20 MEZ, schrieb Daniel Bader <notifications@github.com>:

File: schedule/__init__.py [hunk 1]
```diff
@@ -282,6 +284,18 @@ def do(self, job_func, *args, **kwargs):
         self._schedule_next_run()
         return self
 
+    def persistent(self, identifier, persistent_directory="./*"):
```
Comment (dbader) [inline schedule/__init__.py]: `"./*"` – I think this will blow up on non-UNIX systems.

**Comment by dbader** [comment]
@laubed Not to be rude but I don't think we should move forward with this approach for the reasons stated above. Addressing persistence properly is a fairly large task and I don't think it fits the scope of schedule. If people start using it as a cron replace they're gonna have a bad time. 

However I'd be happy to add an item to the FAQ pointing to the relevant discussions/this PR if you do a quick writeup.

Let's close this PR for now and continue the discussion in a separate issue.

**Pull request closed without merging**